content/en/blog/2024/inpod-traffic-redirection-ambient/index.md

```diff
@@ -0,0 +1,228 @@
+between ztunnel and application pods happens in a way that’s very similar to sidecars and application pods today and is
+strictly invisible to any Kubernetes CNI. Network policy from any Kubernetes CNIs can continue to be enforced,
+regardless of whether the CNI uses eBPF or iptables, there is no conflict between the in-Pod traffic redirection and the CNI.
+
+## Technical deep dive of in-pod traffic redirection
+
+First, let’s go over the basics of how a packet travels between pods in Kubernetes.
+
+### Linux, Kubernetes, and CNI  - What’s A Network Namespace, And Why Does It Matter?
+
+In Linux, a *container* is simply one or more Linux processes running within isolated namespaces, and a Linux namespace
+is simply a kernel flag that controls what processes running within that namespace are able to see. For instance, if you
+create a new Linux network namespace and run a process inside it, that process can only see the networking rules created
+within that network namespace, and not any created outside of it - even though everything running on that machine is still
+sharing one Linux networking stack.
+
+Linux namespaces are conceptually a lot like Kubernetes namespaces - logical labels that organize and isolate different
+active processes, and allow you to create rules about what things within a given namespace can see and what rules are
+applied to them - they simply operate at a much lower level.
+
+When a process running within a network namespace creates a TCP packet outward bound for something else, the packet must be
+processed by any local rules within the local network namespace first, then leave the local network namespace, passing
+into another one.
+
+For example, in plain Kubernetes without any mesh installed, a pod might create a packet and send it to another pod, and
+the packet might (depending on how networking was set up):
+- Be processed by any rules within the source pod’s network namespace.
+- Leave the source pod network namespace, and bubble up into the node’s network namespace where it is processed by any rules in that namespace.
+- From there, finally be redirected into the target pod’s network namespace (and processed by any rules there).
+
+In Kubernetes, the [CRI](https://kubernetes.io/docs/concepts/architecture/cri/) (container *runtime* interface) is responsible for talking to the Linux kernel, creating network namespaces
+for new pods, and starting processes within them. The CRI then invokes the [CNI](https://github.com/containernetworking/cni) (container *networking* interface),
+which is responsible for wiring up the networking rules in the various Linux network namespaces, so that packets leaving and
+entering the new pod can get where they’re supposed to go. It doesn’t matter much what topology or mechanism the CNI uses to
+accomplish this - as long as packets get where they’re supposed to be, Kubernetes works and everyone is happy.
+
+### Istio ambient traffic redirection: why did we drop the previous model?
```

Comment:
I'd remove the "Istio ambient traffic redirection" from the title -- just make it "Why did we drop the previous redirection model?"

Reply:
updated